mlx-macros/tests/test_generate_builder.rs

```diff
@@ -0,0 +1,29 @@
+use mlx_macros::GenerateBuilder;
+
+#[derive(Debug, GenerateBuilder)]
+struct TestStruct {
+    #[optional(default_value = TestStruct::DEFAULT_OPT_FIELD_1)]
```

Comment:
do we support default_value for non optionals? Otherwise it might be simpler for us to just keep the Optional<T> type and just annotate with #[default_value()]